www/js/control.js

```diff
@@ -0,0 +1,431 @@
+'use strict';
+
+angular.module('emission.main.control',['emission.services',
+                                        'emission.splash.startprefs',
+                                        'emission.splash.updatecheck',
+                                        'emission.main.localstorage'])
+
+.controller('ControlCtrl', function($scope, $window, $ionicScrollDelegate,
+               $state, $ionicPopup, $ionicActionSheet, $ionicPopover,
+               $rootScope, StartPrefs,
+               ControlHelper, UpdateCheck, UserCalorieData) {
+    $scope.emailLog = ControlHelper.emailLog;
+    $scope.dark_theme = $rootScope.dark_theme;
+    $scope.userData = []
+    $scope.getUserData = function() {
+        var userDataFromStorage = UserCalorieData.get();
+        $scope.userData = []
+        var height = userDataFromStorage.height.toString();
+        var weight = userDataFromStorage.weight.toString();
+        var temp  =  {
+            age: userDataFromStorage.age,
+            height: height + (userDataFromStorage.heightUnit == 1? ' cm' : ' ft'),
+            weight: weight + (userDataFromStorage.weightUnit == 1? ' kg' : ' lb'),
+            gender: userDataFromStorage.gender == 1? 'Male' : 'Female'
+        }
+        for (var i in temp) {
+            $scope.userData.push({key: i, value: temp[i]});
+        }
+    }
+
+    $scope.userDataSaved = function() {
+        return UserCalorieData.get().userDataSaved == true;
+    }
+    if ($scope.userDataSaved()) {
+        $scope.getUserData();
+    }
+    $scope.getLowAccuracy = function() {
+        //  return true: toggle on; return false: toggle off.
+        if ($scope.settings.collect.config == null) {
+            return false; // config not loaded when loading ui, set default as false
+        } else {
+            var accuracy = $scope.settings.collect.config.accuracy;
+            var v;
+            for (var k in $scope.settings.collect.accuracyOptions) {
+                if ($scope.settings.collect.accuracyOptions[k] == accuracy) {
+                    v = k;
+                    break;
+                }
+            }
+            if ($scope.isIOS()) {
+                return v != "kCLLocationAccuracyBestForNavigation" && v != "kCLLocationAccuracyBest" && v != "kCLLocationAccuracyTenMeters";
+            } else if ($scope.isAndroid()) {
+                return v != "PRIORITY_HIGH_ACCURACY";
+            }
+
```

Comment:
maybe add an `else` here and popup an "unsupported" alert, just to make sure that we remember to extend this if we support other platforms.